--- origsrc/ragel-5.25/test/runtests	2007-12-24 14:15:05.000000000 -0600
+++ src/ragel-5.25/test/runtests	2008-03-24 21:43:56.281250000 -0500
@@ -105,8 +105,8 @@
 
 	# Ruby doesn't need to be compiled.
 	if [ $lang != ruby ]; then
-		echo "$compiler ${cflags} ${out_args} ${code_src}"
-		if ! $compiler ${cflags} ${out_args} ${code_src}; then
+		echo "$compiler ${cflags} ${out_args} ${code_src} ${lflags}"
+		if ! $compiler ${cflags} ${out_args} ${code_src} ${lflags}; then
 			test_error;
 		fi
 	fi
@@ -164,6 +164,7 @@
 			code_suffix=cpp;
 			compiler=$cxx_compiler;
 			cflags="-pedantic -ansi -Wall -O3"
+			lflags=""
 		;;
 		d)
 			lang_opt=-D;
@@ -171,6 +172,7 @@
 			code_suffix=d;
 			compiler=$d_compiler;
 			cflags="-Wall -O3"
+			lflags=""
 		;;
 		c)
 			lang_opt=-C;
@@ -178,13 +180,15 @@
 			code_suffix=c;
 			compiler=$c_compiler;
 			cflags="-pedantic -ansi -Wall -O3"
+			lflags=""
 		;;
 		obj-c)
 			lang_opt=-C;
 			codegen=../rlgen-cd/rlgen-cd;
 			code_suffix=m;
 			compiler=$objc_compiler
-			cflags="-Wall -O3 -fno-strict-aliasing -lobjc"
+			cflags="-Wall -O3 -fno-strict-aliasing"
+			lflags="-lobjc"
 		;;
 		java)
 			lang_opt=-J;
@@ -192,6 +196,7 @@
 			code_suffix=java;
 			compiler=$java_compiler
 			cflags=""
+			lflags=""
 		;;
 		ruby)
 			lang_opt=-R;
@@ -199,6 +204,7 @@
 			code_suffix=rb;
 			compiler=$ruby_engine
 			cflags=""
+			lflags=""
 		;;
 		indep)
 			lang_opt="";
